Dockerfiles/agent/Dockerfile

```
@@ -169,6 +127,47 @@ COPY s6-services /etc/services.d/
 COPY cont-init.d /etc/cont-init.d/
 COPY probe.sh initlog.sh secrets-helper/readsecret.py secrets-helper/readsecret.sh secrets-helper/readsecret_multiple_providers.sh /
 
+##########################################################
+#  Preparation stage: extract Agent package and cleanup  #
+##########################################################
+
+FROM extract-base AS extract
+ARG TARGETARCH
+ARG WITH_JMX
+
+ARG DD_AGENT_ARTIFACT=datadog-agent*-$TARGETARCH.tar.xz
+# copy everything - globbing with args wont work
+COPY ${DD_AGENT_ARTIFACT} /
+WORKDIR /output
+
+# Configuration:
+#   - copy default config files
+COPY datadog*.yaml etc/datadog-agent/
+
+# Installation information
+COPY install_info etc/datadog-agent/
+
+# Extract and cleanup:
+#   - unused systemd unit
+#   - GPL sources for embedded software  # FIXME: move upstream
+#   - docs and manpages                  # FIXME: move upstream
+#   - static libraries                   # FIXME: move upstream
+#   - jmxfetch on nojmx build
+
+RUN --mount=from=artifacts,target=/artifacts \
+ find /artifacts -maxdepth 1 -type f -name "datadog-*-$TARGETARCH.tar.xz" ! -name "$DD_AGENT_ARTIFACT" -exec rm {} \; \
```

why not:
```suggestion
 find /artifacts -maxdepth 1 -type f -name "datadog-*-$TARGETARCH.tar.xz" ! -name "$DD_AGENT_ARTIFACT" -delete \
```
? (like a few line below)